django/core/management/commands/inspectdb.py

```diff
@@ -69,7 +76,21 @@ def strip_prefix(s):
                         constraints = connection.introspection.get_constraints(cursor, table_name)
                     except NotImplementedError:
                         constraints = {}
-                    primary_key_column = connection.introspection.get_primary_key_column(cursor, table_name)
+                    except ValueError as e:
+                        # this might have failed because of sqlite3 handling of views
+                        table_list = connection.introspection.get_table_list(cursor)
```

Comment:
You risk getting this excerpt optimized like this:
```
if any(ti.name == table_name and ti.type == 'v' for ti in connection.introspection.get_table_list(cursor)):
    constraints = {}
else:
    raise e
```
(or similar, modulo indentation) by our idiomatic code wizards :smile: . See e.g. #9493 and #9532

Reply:
@ramiro .. and `else` is unnecessary :smile:

Reply:
ah I knew there would be a more Pythonic way of doing that! Thanks very much. I hope I've got the indentation right.